tests/sqllogictests/suites/base/11_data_type/11_0006_data_type_decimal

```diff
@@ -0,0 +1,114 @@
+query I
+select 1::Decimal(17, 3), 2::Decimal(19, 4), 3::Decimal(20, 5), cast(10 as Decimal(7, 2));
+----
+1.000 2.0000 3.00000 10.00
+
+## parser overflow
+statement error 1005
+select 2::Decimal(2000, 3);
+
+
+## runtime overflow
+statement error 1001
+select 20000::Decimal(4, 3);
+
```

Comment:
should not use code 1001 https://github.com/datafuselabs/databend/blob/main/src/common/exception/src/exception_code.rs#L55

Reply:
This is runtime exception, any error may happen, we don't store the error codes during evaluation for performance issue.

See  `EvalContext:set_error`

Reply:
Will have another pr to use another error code.